components/brave_rewards/browser/rewards_p3a.cc

```diff
@@ -34,76 +38,93 @@ void RecordAutoContributionsState(AutoContributionsP3AState state, int count) {
       answer = 1;
       break;
+    case AutoContributionsP3AState::kWalletCreatedAutoContributeOff:
+      answer = 2;
+      break;
     case AutoContributionsP3AState::kAutoContributeOn:
       switch (count) {
         case 0:
-          answer = 2;
+          answer = 3;
           break;
         case 1:
-          answer = 3;
+          answer = 4;
           break;
         default:
-          answer = 4;
+          answer = 5;
       }
       break;
     default:
       NOTREACHED();
   }
-  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Rewards.AutoContributionsState", answer, 4);
+  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Rewards.AutoContributionsState.2", answer,
+                             5);
 }
 
 void RecordTipsState(bool wallet_created,
+                     bool rewards_enabled,
                      int one_time_count,
                      int recurring_count) {
   DCHECK_GE(one_time_count, 0);
   DCHECK_GE(recurring_count, 0);
 
   int answer = 0;
-  if (wallet_created) {
+  if (wallet_created && !rewards_enabled) {
+    answer = 1;
+  } else if (rewards_enabled) {
+    DCHECK(wallet_created);
     if (one_time_count == 0 && recurring_count == 0) {
-      answer = 1;
+      answer = 2;
     } else if (one_time_count > 0 && recurring_count > 0) {
-      answer = 4;
+      answer = 5;
     } else if (one_time_count > 0) {
-      answer = 2;
-    } else {
       answer = 3;
+    } else {
+      answer = 4;
     }
   }
-  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Rewards.TipsState", answer, 4);
+  UMA_HISTOGRAM_EXACT_LINEAR("Brave.Rewards.TipsState.2", answer, 5);
 }
 
 void RecordAdsState(AdsP3AState state) {
-  UMA_HISTOGRAM_ENUMERATION("Brave.Rewards.AdsState", state);
+  UMA_HISTOGRAM_ENUMERATION("Brave.Rewards.AdsState.2", state);
 }
 
 void UpdateAdsP3AOnPreferenceChange(PrefService *prefs,
                                     const std::string& pref) {
   using brave_rewards::AdsP3AState;
   const bool rewards_enabled =
       prefs->GetBoolean(brave_rewards::prefs::kBraveRewardsEnabled);
+  const bool ads_enabled = prefs->GetBoolean(brave_ads::prefs::kEnabled);
   if (pref == brave_ads::prefs::kEnabled) {
-    if (prefs->GetBoolean(brave_ads::prefs::kEnabled)) {
+    if (ads_enabled) {
       brave_rewards::RecordAdsState(AdsP3AState::kAdsEnabled);
       prefs->SetBoolean(brave_ads::prefs::kAdsWereDisabled, false);
     } else {
       // Apparently, the pref was disabled.
+      // TODO(ifremov): DCHECK(rewards_enabled)?
       brave_rewards::RecordAdsState(
           rewards_enabled ? AdsP3AState::kAdsEnabledThenDisabledRewardsOn :
                             AdsP3AState::kAdsEnabledThenDisabledRewardsOff);
       prefs->SetBoolean(brave_ads::prefs::kAdsWereDisabled, true);
     }
   } else if (pref == brave_rewards::prefs::kBraveRewardsEnabled) {
-    // Rewards pref was changed
+    // Rewards pref was changed.
     if (prefs->GetBoolean(brave_ads::prefs::kAdsWereDisabled)) {
+      DCHECK(!ads_enabled);
```

Comment:
See above comment from @tmancey - if we DCHECK it's best to also then fail gracefully?

Reply:
This is generally a bad practice https://chromium.googlesource.com/chromium/src/+/master/styleguide/c++/c++.md#check_dcheck_and-notreached

we should either write `DCHECK` OR fail gracefully

Reply:
Ah, thanks @iefremov - I hadn't seen this, thanks for clarifying!